### ISSUE TITLE
Reading text-formatted RBR files should be updated

### ISSUE THREAD

**Issue by richardsc** [dankelley/oce#712]
Current support for reading `rsk` files is good (and getting better), but I noticed recently that the facilities for reading ASCII (or "Rtext") formatted RBR files is about 7 years old and doesn't work for newer files. This matters for some older (but not yet obsolete) RBR units that store the raw data using a `.hex` file. The `.hex` file must be exported to an "Rtext" file (engineering units) using Ruskin, and currently oce cannot read such files properly. The reason for this (as always) is that the export format changed when the old Windows-only software was replaced with Ruskin.

I propose to add the functionality to be able to read such files (I know of at least one data repository interested in being able to read these files into R). I will work in the `rsktxt` branch so as not to collide with `develop`, and will merge when necessary.

A few things to think about:

1. Since these aren't *actually* rsk files, it seems a little strange to make them `rsk` objects. However, this is what we currently do for the TDR txt files that we currently support, and the alternative is to invent another new class. I therefore propose that such files should be handled by `read.rsk()` and be given the `rsk` class.

2. Should the current code (which can read the "old-style" text format files) be kept? If the answer to this is "No", then this would affect the current `rsk` dataset, which is based on an "old" style TDR file. Since we had already been talking about replacing the `rsk` dataset with an *actual* rsk file, I don't think this would be a big deal in the long run.

Thoughts, @dankelley?

**Comment by richardsc** [comment]
Also, I'll be putting some data and code into `oce-issues/712`, but the data files are several MB. I won't push them until I'm sure I want to commit the size, and that the files are suitable for Github.

**Comment by richardsc** [comment]
Tasks:

* [x] make `oceMagic()` recognize the file type
* [ ] make `read.oce()` recognize the file type, and pass `type='txt'` to `read.rsk()`
* [x] add case to `read.rsk()` for `type=txt`
* [ ] make all args work for `read.rsk(..., type='txt')` e.g. `from=`, `to=`, ...
* [ ] update `read.rsk()` docs
* [ ] update `rsk-class` docs for txt format files

**Comment by dankelley** [comment]
Thanks for this! Re your questions...

1. "Since these aren't actually rsk files..." I agree, call them rsk, since that seems the closest.
2. "Should the current code (which can read the "old-style" text format files) be kept?" I think it should be deleted, actually. Anyone (like us) who wants to use the old style could just use ``read.table()`` by themselves, which would be fine, I would argue. Generally, I think the ``rsk`` dataset should be something that users are likely to have, i.e. they should be from a popular instrument.

**Comment by richardsc** [comment]
Ok, I think I've got everything in there. @dankelley I'd appreciate it if you could take a look at the `rsktxt` branch as well as some of the tests in [oce-issues/712](https://github.com/dankelley/oce-issues/tree/f777ceb339d6c0baad3198fd9405c8c279498907/712) to see if I've done things halfway sensibly. I haven't done a check `--as-cran` so you might want to run things through your regular build/check to see if I've done anything Ripleyable. 

I agree that the legacy code (which really only worked for the sleiwex tdr files) should be removed. However, I wanted to make sure I had the new code working first. In adding the new code I found things that I'm pretty sure were wrong in the old code, but were never detected because we weren't using it often enough. Removing the old code will require changes to a number of functions, including:
* `read.rsk()`
* `oceMagic()`
* `read.oce()`
* `create_data/rsk/create_rsk.R` (e.g. the example tdr dataset)
* a number of man pages, especially `read.rsk.Rd`

**Comment by dankelley** [comment]
Thanks. It builds for me, and the tests in 712 work except for a lot of grep warnings like
```
1: In grep("Model", l) : input string 17 is invalid in this locale
```
which I will fix.  (Probably I should set a fancier locale on my machine but I think its smarter to have oce working for people without fancy locale settings.)

**Comment by dankelley** [comment]
I've fixed up the ``grep`` problems that came up in 712. I suppose there may be others that we'll notice one day.  Well, maybe it's only I that will notice them, since I have a non-fancy locale. I'm surprised that ``R CMD check`` doesn't set to such a locale, actually.

PS. these are quite confusing dates in the old pt file
```
$ grep time pt_rbr_014765.dat 
Host time     08/07/02 14:22:00
Logger time   08/07/02 14:22:04
```
so I can see why they wanted to update that format! I don't think I'll ever understand how people learn to read dates when written in such formats. Maybe in a non-technical context they already know the year, and they hope for a number greater than 12 to see which other part is a day or something.

**Comment by dankelley** [comment]
As for the legacy code, I propose we leave it in there until we get an official new ``data(rsk)``, ideally also with a datafile we can ship in ``inst/extdata``, which means it should be about 40K, to be in the ballpark of the others. Certainly this (very nice) 1.3M file is way too big (it's 429K even when gzipped).

I'd also propose that I remove the legacy code, because then my eyes will be going over the new code also, and that will be a double check on things ... not that I see anything (except the locale part) wrong with the nice code you put in the branch.

To be honest, I don't see any reason not to merge this into develop, but if you had the idea that we might get a new datafile soon, we could wait.  (I just worry about branches that sit for a long time ... something silly can always come up to make a merge conflict when things are left for months.)

**Comment by dankelley** [comment]
Just for fun, I tried making the ``0*.txt`` file into ``rsk.txt``; see the ``README.md`` for details. The unzipped file is still too long for oce, but I don't see any reason why a lot at the start cannot be trimmed. Pic is below. 

<img width="680" alt="screen shot 2015-08-12 at 9 33 56 am" src="https://cloud.githubusercontent.com/assets/99469/9223987/6076d9fc-40d5-11e5-923d-6a22bba4d8cb.png">

**Comment by dankelley** [comment]
I made a new ``rsk2.txt`` that's shorter, by trimming out most of the equilibration time; again, see the README. @richardsc -- I think I'm done for a while. The next step, which I hope you can take, is to decide on a new rsk dataset.  I can do the actual addition (which is slightly funky, e.g. the entry in ``oce/create_data`` should be updated, even though that is hidden from the actual oce package by the ``.Rbuildignore`` directive).

**Comment by richardsc** [comment]
Thanks, I'll have a look. I agree that the old code can be removed once we've got a suitable example file for a replacement `rsk` data set. I have a few ideas for this which I'll document in a new issue (#713).

I'll close this once I've merged with develop and am satisfied that things are working.

**Comment by richardsc** [comment]
Merged with PR #714. Closing.
